infra/src/main/kotlin/fi/fta/geoviite/infra/geometry/GeometryDao.kt

```diff
@@ -626,90 +635,146 @@ class GeometryDao @Autowired constructor(
+            ) alignments on (true)
+          where (:plan_id::int is null or (:plan_id = plan.id and :plan_version = plan.version))
         """.trimIndent()
         val params = mapOf(
             "plan_id" to rowVersion.id.intValue,
+            "plan_version" to rowVersion.version,
         )
-        val header = jdbcTemplate.queryForObject(sql, params) { rs, _ ->
-            val project = Project(
-                id = rs.getIntId("project_id"),
-                name = ProjectName(rs.getString("project_name")),
-                description = rs.getFreeTextOrNull("project_description"),
-            )
+        return getOne(rowVersion.id, jdbcTemplate.query(sql, params) { rs, _ -> getPlanHeader(rs) }).also { header ->
+            logger.daoAccess(FETCH, GeometryPlanHeader::class, header.version)
+        }
+    }
 
-            val minKm = rs.getKmNumberOrNull("min_km_number")
-            val maxKm = rs.getKmNumberOrNull("max_km_number")
-            val range = if (minKm != null && maxKm != null) Range(minKm, maxKm) else null
-            GeometryPlanHeader(
-                id = rs.getIntId("plan_id"),
-                project = project,
-                fileName = rs.getFileName("file_name"),
-                source = rs.getEnum("source"),
-                kmNumberRange = range,
-                planTime = rs.getInstantOrNull("plan_time"),
-                trackNumberId = rs.getIntIdOrNull("track_number_id"),
-                measurementMethod = rs.getEnumOrNull<MeasurementMethod>("measurement_method"),
-                elevationMeasurementMethod = rs.getEnumOrNull<ElevationMeasurementMethod>("elevation_measurement_method"),
-                decisionPhase = rs.getEnumOrNull<PlanDecisionPhase>("plan_decision"),
-                planPhase = rs.getEnumOrNull<PlanPhase>("plan_phase"),
-                message = rs.getFreeTextOrNull("message"),
-                linkedAsPlanId = rs.getIntIdOrNull("linked_as_plan_id"),
-                uploadTime = rs.getInstant("upload_time"),
-                units = GeometryUnits(
-                    coordinateSystemSrid = rs.getSridOrNull("srid"),
-                    coordinateSystemName = rs.getString("coordinate_system_name")?.let(::CoordinateSystemName),
-                    verticalCoordinateSystem = rs.getEnumOrNull<VerticalCoordinateSystem>("vertical_coordinate_system"),
-                    directionUnit = rs.getEnum("direction_unit"),
-                    linearUnit = rs.getEnum("linear_unit"),
-                ),
-                author = rs.getString("author"),
-                hasProfile = rs.getBoolean("has_profile"),
-                hasCant = rs.getBoolean("has_cant"),
-                isHidden = rs.getBoolean("hidden"),
-            )
-        } ?: throw NoSuchEntityException(GeometryPlanHeader::class, rowVersion.id)
-        logger.daoAccess(FETCH, GeometryPlanHeader::class, rowVersion.id)
-        return header
+    private fun getPlanHeader(rs: ResultSet): GeometryPlanHeader {
```

Comment:
En oikein tykkää pistää näitä resultsetmappauksia erillisiin funkkareihin koska se eriyttää sql:n column-nimet tästä käyttöpaikasta... mutta näissä preload vs fetch funkkareissa tuo sisältö on siinä määrin räikeä duplikaatti että irrotin ne nyt sitten kuitenkin.